### DISCUSSION TITLE
Task 2 Comma error

### DISCUSSION THREAD

**Discussion by mmh62** [Part-II-Computational-Physics/.github#50] [General]
This is my code for task 2: 

```
def CompleteHamiltonian2(N, h=1.):
    H = np.zeros((2*N, 2*N))
    Hloc =  - ft.reduce(np.kron, [sigmaZ, sigmaZ]) - 0.5*h*(ft.reduce(np.kron, [sigmaX,I]) *h*ft.reduce(np.kron, [ I, sigmaX]))
    for i in range(N-1):
        H +=(np.kron,[np.eye(2*i), Hloc, np.eye(2*(N-2-i))])
    H = H - (np.kron,[sigmaZ, np.eye(2*(N-2)), sigmaZ]) - (h/2) * (np.kron(sigmaX, np.eye(2(N-1))) + np.kron(np.eye(2*(N-1)), sigmaX))
    
    return H
```
    

It gives this error: 

<>:7: SyntaxWarning: 'int' object is not callable; perhaps you missed a comma?
<>:7: SyntaxWarning: 'int' object is not callable; perhaps you missed a comma?

Please help

**Comment by JNHallen** [comment]
First a general remark: Try not to use ft.reduce if you are only applying the function once. `ft.reduce(np.kron, [sigmaZ, sigmaZ])` does the exact same thing as `np.kron(sigmaZ, sigmaZ)`. 

I suspect your error comes from the fact that you are trying to do `(np.kron,[sigmaZ, np.eye(2*(N-2)), sigmaZ])` when you actually want to do either` np.kron(sigmaZ, np.kron( np.eye(2*(N-2)), sigmaZ))` or `ft.reduce(np.kron,[sigmaZ, np.eye(2*(N-2)), sigmaZ])`. 

You seem to be multiplying your sigmaX operators in Hloc instead of adding them.
